### PR TITLE
Zsd node thread

### PR DESCRIPTION
**Pull request by zsd-planekid**
Reformatted nodes and added node watchdog

### CODE REVIEW

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/actuators/brake.java [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-	/***
-	 * the system attempts to brake, if this fails then a fatal error will be
-	 * raised
-	 */
-	public void releaseBrake() {
-		desiredState = false;
-		// TODO implment
-	}
-
-}
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/actuators/brake.java]: I want us to keep a safety/brake node in some sense.  I think it should be a periodic node that will run in the background and cause an estop if the system ever determines that something is wrong.

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/Robot.java [hunk 2]
```diff
@@ -1,27 +1,13 @@
 package com.roboclub.robobuggy.main;
 
-import java.util.ArrayList;
+import java.util.LinkedList;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/Robot.java]: Is their a specific reason why you want linkedList over arrayList?

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java [hunk 3]
```diff
@@ -1,24 +1,19 @@
-package com.roboclub.robobuggy.nodes;
+package com.roboclub.robobuggy.nodes.sensors;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java]: Make the Logging Node actually Log stuff

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 4]
```diff
@@ -167,7 +141,7 @@ public int matchDataMinSize()
 	//must be the same as the baud rate of the arduino 
 	public int baudRate()
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: change to getBaudRate

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 5]
```diff
@@ -272,20 +248,72 @@ else if (sensorName.equals("encoder"))
...
+			return true;
+		}
+		
 	}
 	
 	/**
 	 * Private class used to represent RBSM messages
 	 */
-	private class RBSMessage {
+	private static class RBSMessage {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: ?????

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/SensorPlayer.java [hunk 6]
```diff
@@ -182,7 +182,7 @@ public void run() {
 						break;
 						
 					default:
-						new RobobuggyLogicException("Found an unsupported sensor in the logs!", MessageLevel.WARNING);
+						//new RobobuggyLogicException("Found an unsupported sensor in the logs!", MessageLevel.WARNING);
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/SensorPlayer.java]: @trevor look into this shutting down the system

Comment (trevor) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/SensorPlayer.java]: Shut it all down.

Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/SensorPlayer.java]: oops wrong Trevor, but I agree we should shut everything down

**Comment by TrevorDecker** [comment]
This has been tested and is ready to merge

**Pull request merged**